### PR TITLE
#408 if condition around $pathQualifier

### LINKED ISSUE #408: win_acl_inheritance does not support UNC path with the recent commit
##### SUMMARY
win_acl_inheritance checks for a path qualifier. Path qualifier is not supported on a network path.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
win_acl_inheritance

##### ANSIBLE VERSION
```
ansible [core 2.11.12]
```

##### COLLECTION VERSION
```
ansible.windows 1.11.1
```

##### CONFIGURATION
```
ANSIBLE_FORCE_COLOR(env: ANSIBLE_FORCE_COLOR) = True
CALLBACKS_ENABLED(env: ANSIBLE_CALLBACK_WHITELIST) = ['profile_tasks', 'timer', 'junit']
COLLECTIONS_PATHS(/etc/ansible/ansible.cfg) = ['/root/.ansible/collections', '/projdev/collections', '/usr/share/ansible/collections']
DEFAULT_ACTION_PLUGIN_PATH(env: ANSIBLE_ACTION_PLUGINS) = ['/usr/share/venv/lib/python3.7/site-packages/ara/plugins/action']
DEFAULT_CALLBACK_PLUGIN_PATH(env: ANSIBLE_CALLBACK_PLUGINS) = ['/usr/share/venv/lib/python3.7/site-packages/ara/plugins/callback']
DEFAULT_FORKS(/etc/ansible/ansible.cfg) = 50
DEFAULT_LOAD_CALLBACK_PLUGINS(/etc/ansible/ansible.cfg) = True
DEFAULT_LOOKUP_PLUGIN_PATH(env: ANSIBLE_LOOKUP_PLUGINS) = ['/usr/share/venv/lib/python3.7/site-packages/ara/plugins/lookup']
DEFAULT_MODULE_PATH(env: ANSIBLE_LIBRARY) = ['/Users/MYUSERNAME/library']
DEFAULT_MODULE_UTILS_PATH(/etc/ansible/ansible.cfg) = ['/root/.ansible/module_utils', '/opt/projcs/ansible/ansible_plugins/module_utils', '/projdev/module_utils']
DEFAULT_ROLES_PATH(env: ANSIBLE_ROLES_PATH) = ['/Users/MYUSERNAME/git/roles']
DEFAULT_VERBOSITY(env: ANSIBLE_VERBOSITY) = 3
DEPRECATION_WARNINGS(/etc/ansible/ansible.cfg) = False
HOST_KEY_CHECKING(env: ANSIBLE_HOST_KEY_CHECKING) = False
```

##### OS / ENVIRONMENT
VMware Photon OS 3.0

##### STEPS TO REPRODUCE
Specify `path` parameter as any UNC network path instead of a local filesystem path.

```
- name: Remove Inheritence on UNC path
   win_acl_inheritance_ws1cs:
     path: \\amznfsx8sjdj59.my.lab\D$\Path\Customer
     state: absent
```

##### EXPECTED RESULTS
Skip splitting of the path qualifier and finish setting inheritance as needed

##### ACTUAL RESULTS
```
The full traceback is:
You cannot call a method on a null-valued expression.
At line:98 char:1
+ $pathQualifier = $pathQualifier.Replace(':', '')
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : InvokeMethodOnNull

ScriptStackTrace:
at <ScriptBlock>, <No file>: line 98
fatal: [servername.my.lab]: FAILED! => {
    "changed": false,
    "msg": "Unhandled exception while executing module: You cannot call a method on a null-valued expression."
}
```

### PR DESCRIPTION
**Pull request by aaronk1**
##### SUMMARY
If statement around $pathQualifier `:` in case the `path` parameter is a UNC path

Fixes #408 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
win_acl_inheritance

##### ADDITIONAL INFORMATION

Before
```
New-PSDrive : Cannot bind argument to parameter 'Name' because it is null.
At line:12 char:31
+     $null = New-PSDrive -Name $pathQualifier -PSProvider 'Registry' - ...
+                               ~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [New-PSDrive], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.NewPSDriveCommand
```

After
```
```

In testing, this completes successfully and the module runs as expected.

### CODE REVIEW

**Comment by aaronk1** [comment]
@briantist please review. Removed the old PR as I messed up a few things accidentally and couldn't squash the ~10 commits. Should all be fixed now. Hoping for successful builds 🤞

File: plugins/modules/win_acl_inheritance.ps1 [hunk 1]
```diff
@@ -95,7 +95,9 @@ $regeditHives = @{
 }
 
 $pathQualifier = Split-Path -Path $path -Qualifier -ErrorAction SilentlyContinue
-$pathQualifier = $pathQualifier.Replace(':', '')
+if ($pathQualifier) {
```
Comment (briantist) [inline plugins/modules/win_acl_inheritance.ps1:98]:
```suggestion
# UNC paths will not return a qualifier, so $pathQualifier will be $null in that case
if ($pathQualifier) {
```

**Review by briantist — COMMENTED** [review]
The change looks good to me, the only thing I'd like to see is an addition for UNC paths. If those were in the tests the issue would have been caught on the original change. Adding them now will prevent regressions.

The tests can be found in `tests/integration/targets/win_acl_inheritance` and they are written in Ansible.

Would you be up for trying to add a test there?

**Review by jborean93 — CHANGES_REQUESTED** [review]
Are you able to add a changelog fragment as documented under https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment to document this bugfix please.

**Comment by aaronk1** [comment]
> Are you able to add a changelog fragment as documented under https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment to document this bugfix please.

added

**Comment by aaronk1** [comment]
> The change looks good to me, the only thing I'd like to see is an addition for UNC paths. If those were in the tests the issue would have been caught on the original change. Adding them now will prevent regressions.
> 
> The tests can be found in `tests/integration/targets/win_acl_inheritance` and they are written in Ansible.
> 
> Would you be up for trying to add a test there?

Sure. May be a little until I can get to it though.

**Comment by jborean93** [comment]
Just an FYI you can probably just change one of the existing tests so the path is `\\localhost\c$\...`. The hardest bit will be figuring out how to convert the local path to be `\\localhost\c$` but that shouldn't be too hard to achieve.

**Comment by jborean93** [comment]
Will look at adding the tests in the future, for now it will be good to have this change in place.

**Pull request merged**